.github/workflows/build-all-site.yml

```diff
@@ -0,0 +1,42 @@
+name: PR Build documentation site
+on:
+  pull_request:
+    # To manage 'surge-preview' action teardown, add default event types + closed event type
+    types: [opened, synchronize, reopened, closed]
```

Comment:
we don't need to watch closed here, closed is used to let surge teardown the deployment and we don't use surge here